app/src/main/java/org/mozilla/fenix/wallpapers/WallpaperManager.kt

```diff
@@ -122,6 +131,8 @@ class WallpaperManager(
             wallpapers.find { it.name == currentWallpaper }
                 ?: fileManager.lookupExpiredWallpaper(currentWallpaper)
                 ?: defaultWallpaper
+        }.also {
+            appStore.dispatch(AppAction.WallpaperAction.UpdateCurrentWallpaper(it))
```

Comment:
Why do we need to do this here when you do it in the setter of `currentWallpaper`?

Reply:
Setters aren't called during initialization, which is the only time this is called. There's another weird instance of it being called in `switchToNextWallpaper`, where `currentWallpaper` isn't actually set. In both cases, `currentWallpaper` is being set during `updateWallpaper`, which is called from client code. These are meant as fallbacks since it felt like bad practice to assume client behavior.

In any case, `updateWallpaper` will be removed in a follow-up PR, and a lot of these oddities will be cleaned up then. Trying to practice breaking patches into small pieces, and you can see some other updates in the pipeline on the [parent issue](https://github.com/mozilla-mobile/fenix/issues/26034)